source/javascripts/services/_stepSourceService.js.erb

```diff
@@ -49,20 +49,26 @@
 				var stepVersions = library.steps[step.id];
 				var shouldLoadOnlyStepIcon = !requestedVersion || !(stepVersions && stepVersions[requestedVersion]);
 
-				if (shouldLoadOnlyStepIcon) {
-					requestedVersion = library.latestStepVersions[step.id];
-					step.assetUrls(library.steps[step.id][requestedVersion].defaultStepConfig.asset_urls);
-				} else {
-					step.version = semverService.normalizeVersion(step.version);
-					step.defaultStepConfig = library.steps[step.id][requestedVersion].defaultStepConfig;
-				}
+				var logParams = {
+					id: step.id,
+					version: step.version,
+					stepsLength: Object.keys(library.steps).length
+				};
 
-				if (!step.isConfigured()) {
-					logger.warn("Step is not configured", {
-						id: step.id,
-						version: step.version,
-						stepsLength: Object.keys(library.steps).length
-					});
+				try {
+					if (shouldLoadOnlyStepIcon) {
+						requestedVersion = library.latestStepVersions[step.id];
+						step.assetUrls(library.steps[step.id][requestedVersion].defaultStepConfig.asset_urls);
+					} else {
+						step.version = semverService.normalizeVersion(step.version);
+						step.defaultStepConfig = library.steps[step.id][requestedVersion].defaultStepConfig;
+					}
+
+					if (!step.isConfigured()) {
+						logger.warn("Step is not configured", logParams);
+					}
+				} catch (error) {
+					logger.error(error, logParams);
```

Comment:
we still don't have enough context on this issue error seen in datadog: `undefined is not an object: library.steps[step.id][requestedVersion]` so step.id for some reason does not exist in steps